snapcraft/plugins/qbs.py

```diff
@@ -0,0 +1,157 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2017 Dan Chapman <dpniel@ubuntu.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""The QBS plugin is useful for building parts that use the Qt Build Suite.
+
+These are projects that are built using .qbs files and are using gcc or
+clang based toolchains.
+
+This plugin uses the common plugin keywords as well as those for "sources".
+For more information check the 'plugins' topic for the former and the
+'sources' topic for the latter.
+
+Additionally, this plugin uses the following plugin-specific keywords:
+
+    - qbs-options:
+      (list of strings)
+      Additional options to pass to the qbs invocation.
+    - qt-version:
+      (enum, 'nil' or 'qt4' or 'qt5')
+      Version of Qt to use for qbs profile. If nil the base gcc/clang profile
+      will be used.
+    - qbs-profile:
+      (enum, 'gcc' or 'clang')
+      Set either gcc or clang as the base build profile. Default is gcc
+    - qbs-build-variant:
+      (enum, 'debug' or 'release')
+      Set either debug or release build variant. Default is release.
+"""
+import os
+import shutil
+import snapcraft
+
+
+class QbsPlugin(snapcraft.BasePlugin):
+
+    @classmethod
+    def schema(cls):
+        schema = super().schema()
+        schema['properties']['qbs-options'] = {
+            'type': 'array',
+            'minitems': 1,
+            'uniqueItems': True,
+            'items': {
+                'type': 'string',
+            },
+            'default': [],
+        }
+
+        schema['properties']['qbs-profile'] = {
+            'enum': ['gcc', 'clang'],
+            'default': 'gcc',
+        }
```

Comment:
Heh, that's cool! I'm happy to get this right and learn how you guys like it done :-)